### PR TITLE
Update regex for pobox filter

### PR DESCRIPTION
**Pull request by Gbarra9**
This pull request is related to:
- https://github.com/episphere/connect/issues/1063
- https://github.com/episphere/connect/issues/1118

Code change: 
- update po box regex to handle "pobox" followed by optional digits.

### CODE REVIEW

**Review by bransteitterbr — APPROVED** [review]

File: utils/firestore.js [hunk 1]
```diff
@@ -2279,7 +2279,7 @@ const processParticipantHomeMouthwashKitData = (record, printLabel) => {
     const { collectionDetails, baseline, bioKitMouthwash, firstName, lastName, address1, address2, city, state, zip } = fieldMapping;
 
     const addressLineOne = record?.[address1];
-    const poBoxRegex = /\b(?:P\.?O\.?(?:\s*Box|\s+Office\s+Box)|Post\s+Office\s+Box)\b/i;
+    const poBoxRegex = /\b(?:P\.?O\.?(?:\s*Box|\s+Office\s+Box)|Post\s+Office\s+Box|pobox)\d*\b/i;
```
Comment (we-ai) [inline utils/firestore.js:2282]: "po office box" can pass this regex, which may not be ideal.
An updated regex can be: `/^(?:P\.?O\.?\s*Box|Post\s+Office\s+Box)\s+\d+$/i`

**Review by we-ai — COMMENTED** [review]
`\d*` is not ideal: allowing 0 digit. Also, not sure "P.O. Office Box" is an intended input, but allowed by the regex.

File: utils/firestore.js (hunk 1 above)
Comment (Gbarra9) [reply utils/firestore.js:2282]: Makes sense, thanks for the suggestion. I'll have to modify the regex to include numbers as well as these other "po box" variations that are being inputted

Comment (Gbarra9) [reply utils/firestore.js]: @we-ai I plan to use this regex to handle other variations, some variations already exist in Prod. All are returning true.

`//P.O Box:

"P.O Box#123"
"P.O Box 123"
"P.O Box #123"
"P.O Box123 Main st"

//P.O. Box:

"P.O. Box123"
"P.O. Box#123"
"P.O. Box 123"
"P.O. Box #123"


//POBox:

"POBox123"
"POBox#123"
"POBox 123"
"POBox #123"


//po box:

"po box123"
"po box#123"
"po box 123"
"po box #123"

//PO BOX:

"PO BOX123"
"PO BOX#123"
"PO BOX 123"
"PO BOX #123"

//Po Box:

"Po Box123"
"Po Box#123"
"Po Box 123"
"Po Box #123"


//pob:

"pob123"
"pob#123"
"pob 123"
"pob #123"


//p.o.b.:

"p.o.b.123"
"p.o.b.#123"
"p.o.b. 123"
"p.o.b. #123"
`

Comment (we-ai) [reply utils/firestore.js]: The updated regex could  be simpler by removing duplicates: `/^(?:P\.?O\.?\s*(?:Box|B\.?)?|Post\s+Office\s+(?:Box|B\.?)?)\s*#?\s*\d+(\s+(.+)?$/i`. I didn't test this.
"p.oBox #123" can also pass, just let you know.

**Review by we-ai — APPROVED** [review]
Looks good.

**Comment by Gbarra9** [comment]
@we-ai @bransteitterbr I updated the regex to make numbers optional. There are currently addresses with no number after box.

**Pull request merged**